homeassistant/components/template/light.py

```diff
@@ -234,30 +227,14 @@ def __init__(
         self._supports_transition = False
         self._supported_color_modes = None
 
-        color_modes = {ColorMode.ONOFF}
-        if self._level_script is not None:
-            color_modes.add(ColorMode.BRIGHTNESS)
-        if self._temperature_script is not None:
-            color_modes.add(ColorMode.COLOR_TEMP)
-        if self._hs_script is not None:
-            color_modes.add(ColorMode.HS)
-        if self._color_script is not None:
-            color_modes.add(ColorMode.HS)
-        if self._rgb_script is not None:
-            color_modes.add(ColorMode.RGB)
-        if self._rgbw_script is not None:
-            color_modes.add(ColorMode.RGBW)
-        if self._rgbww_script is not None:
-            color_modes.add(ColorMode.RGBWW)
-
         self._supported_color_modes = filter_supported_color_modes(color_modes)
         if len(self._supported_color_modes) > 1:
             self._color_mode = ColorMode.UNKNOWN
         if len(self._supported_color_modes) == 1:
             self._color_mode = next(iter(self._supported_color_modes))
```

Comment:
Can we keep this validation of color modes together with how the supported color modes are determined?
Maybe move this loop to here?
```py
        color_modes = {ColorMode.ONOFF}
        for action_id, color_mode in (...
            ...
        self._supported_color_modes = filter_supported_color_modes(color_modes)
```